app/routines/user_from_signup_state.rb

```diff
@@ -0,0 +1,22 @@
+class UserFromSignupState
+
+  lev_routine express_output: :user
+
+  protected
+
+  def exec(signup_state)
+    user = User.new
+    if signup_state.trusted?
+      user.role = signup_state.role
+      user.full_name = signup_state.trusted_data['name']
+      user.self_reported_school = signup_state.trusted_data['school']
+      if signup_state.trusted_data['uuid']
+        user.external_uuids.build(uuid: signup_state.trusted_data['uuid'])
+      end
+    end
+    user.save
+    outputs.user = user
+    transfer_errors_from(user, {type: :verbatim}, true)
+  end
```

Comment:
@jpslav I wonder if we should add a json field to user and store the trusted data there so it's retained?  As we'd discussed we're setting the `self_reported_school` to something that's not really self_reported, instead it's trusted.  

Down the road it might be nice if we tell apart what was trusted and self-reported,

Reply:
I like that idea.